src/golang/lib/workflow/dag/workflow_dag.go

```diff
@@ -111,33 +102,31 @@ func computeArtifactSignatures(
 			bytesToHash = append(bytesToHash, []byte(currOp.Spec.Param().Val)...)
 		}
 
-		if len(currOp.Outputs) > 1 {
-			return nil, errors.Newf("Multiple output artifacts is currently unsupported.")
-		}
-
 		// Compute that signature for each output artifact.
 		// The assumption is that there is only one output.
-		outputArtifactID := currOp.Outputs[0]
-		bytesToHash = append(bytesToHash, []byte(outputArtifactID.String())...)
-
-		// Compute that final hash and add it to the map, then continue traversing.
-		artifactIDToSignature[outputArtifactID] = uuid.NewSHA1(uuid.NameSpaceOID, bytesToHash)
-		processedArtifactIds[outputArtifactID] = true
-
-		// Find the next downstream operators. We must have already visited all the operator's inputs.
-		for _, nextOpID := range opIDsByInputArtifact[outputArtifactID] {
-			nextOp := dbOperators[nextOpID]
-
-			depsComputed := true
-			for _, inputArtifactID := range nextOp.Inputs {
-				if _, ok := processedArtifactIds[inputArtifactID]; !ok {
-					depsComputed = false
-					break
+		for _, outputArtifactID := range currOp.Outputs {
+			bytesToHash = append(bytesToHash, []byte(outputArtifactID.String())...)
```

Comment:
maybe doesn't affect correctness, but every loop I think we should be appending each output artifact id to the `bytesToHash` in line 102. In the current implementation for example the second output artifact's signature seems to contain the first output artifact's id, which is a bit weird.

Reply:
oh yeah you're totally right - will fix.